R/significance_annotation.R

```diff
@@ -227,12 +230,35 @@ GeomSignif <- ggplot2::ggproto("GeomSignif", ggplot2::Geom,
                                              vjust = 0, alpha = NA, family = "", fontface = 1, lineheight = 1.2, linetype=1, size=0.5),
                            draw_key = function(...){grid::nullGrob()},
 
-                           draw_group = function(data, panel_params, coord, parse=FALSE) {
+                           draw_group = function(data, panel_params, coord, parse=FALSE, extend_line = extend_line) {
                              lab <- as.character(data$annotation)
                              if (parse) {
                                lab <- parse_safe(as.character(lab))
                              }
                              coords <- coord$transform(data, panel_params)
+                             if ( extend_line != 0 & nrow(coords) == 3 ) {
+                               if ( coords[2,'x'] < coords[2,'xend'] ) {
+                                # left vertical segment
+                                coords[1,'x'] <- coords[1,'x'] - extend_line
+                                coords[1,'xend'] <- coords[1,'xend'] - extend_line
+                                # horizontal line
+                                coords[2,'x'] <- coords[2,'x'] - extend_line
+                                coords[2,'xend'] <- coords[2,'xend'] + extend_line
+                                # right vertical segment
+                                coords[3,'x'] <- coords[3,'x'] + extend_line
+                                coords[3,'xend'] <- coords[3,'xend'] + extend_line
+                               } else if ( coords[2,'x'] > coords[2,'xend'] ) {
+                                # left vertical segment
+                                coords[1,'x'] <- coords[1,'x'] + extend_line
+                                coords[1,'xend'] <- coords[1,'xend'] + extend_line
+                                # horizontal line
+                                coords[2,'x'] <- coords[2,'x'] + extend_line
+                                coords[2,'xend'] <- coords[2,'xend'] - extend_line
+                                # right vertical segment
+                                coords[3,'x'] <- coords[3,'x'] - extend_line
+                                coords[3,'xend'] <- coords[3,'xend'] - extend_line
```

Comment:
Do you think it is worth shortening this part by testing once in the beginning `if(coords[2,'x'] > coords[2,'xend'] )` and then do `extend_line <-  - extend_line`?

Reply:
That's pretty clever!

Reply:
Perfect